python/whylogs/core/constraints/metric_constraints.py

```diff
@@ -171,19 +185,109 @@ def validate_profile(self, dataset_profile: DatasetProfileView) -> Tuple[bool, O
         return (validate_result, metric_summary)
 
 
+@dataclass
+class DatasetConstraint:
```

Comment:
Look at pulling out a common base class with `MetricConstraint`

Reply:
The `name` and `require_column_existence` are common. `validate_profile()` could be an abstract method. I think I vote to hold off on extracting a base class until we need another would-be subclass...